pydicom/tests/test_overlay_np.py

```diff
@@ -0,0 +1,764 @@
+# JPEG2k
+JPEG_2K = get_testdata_files("JPEG2000.dcm")[0]
+# RLE Lossless
+RLE = get_testdata_files("MR_small_RLE.dcm")[0]
+
+# Transfer Syntaxes (non-retired + Explicit VR Big Endian)
+SUPPORTED_SYNTAXES = [
+    ImplicitVRLittleEndian,
+    ExplicitVRLittleEndian,
+    DeflatedExplicitVRLittleEndian,
+    ExplicitVRBigEndian
+]
+UNSUPPORTED_SYNTAXES = list(
+    set(ALL_TRANSFER_SYNTAXES) ^ set(SUPPORTED_SYNTAXES)
+)
+
+
+def test_unsupported_syntaxes():
+    """Test that UNSUPPORTED_SYNTAXES is as expected."""
+    for syntax in SUPPORTED_SYNTAXES:
+        assert syntax not in UNSUPPORTED_SYNTAXES
+
+
+REFERENCE_DATA_UNSUPPORTED = [
+    (JPEG_BASELINE_1, ('1.2.840.10008.1.2.4.50', 'Lestrade^G')),
+    (JPEG_EXTENDED_2, ('1.2.840.10008.1.2.4.51', 'CompressedSamples^NM1')),
+    # (JPEG_LOSSLESS_14, ('1.2.840.10008.1.2.4.57')),  # No dataset available
+    (JPEG_LOSSLESS_14_1, ('1.2.840.10008.1.2.4.70', 'Lestrade^G')),
+    (JPEG_LS_LOSSLESS, ('1.2.840.10008.1.2.4.80', 'CompressedSamples^MR1')),
+    # (JPEG_LS_LOSSY, ('1.2.840.10008.1.2.4.81')),  # No dataset available
+    (JPEG_2K_LOSSLESS, ('1.2.840.10008.1.2.4.90', '')),
+    (JPEG_2K, ('1.2.840.10008.1.2.4.91', 'CompressedSamples^NM1')),
+    (RLE, ('1.2.840.10008.1.2.5', 'CompressedSamples^MR1')),
+]
+
+
+# Numpy and the numpy handler are unavailable
+@pytest.mark.skipif(HAVE_NP, reason='Numpy is available')
+class TestNoNumpy_NoNumpyHandler(object):
+    """Tests for handling datasets without numpy and the handler."""
+    def setup(self):
+        """Setup the environment."""
+        self.original_handlers = pydicom.config.overlay_data_handlers
+        pydicom.config.overlay_data_handlers = []
+
+    def teardown(self):
+        """Restore the environment."""
+        pydicom.config.overlay_data_handlers = self.original_handlers
+
+    def test_environment(self):
+        """Check that the testing environment is as expected."""
+        assert not HAVE_NP
+        assert NP_HANDLER is not None
+
+    def test_can_access_supported_dataset(self):
+        """Test that we can read and access elements in dataset."""
+        # Explicit little
+        ds = dcmread(EXPL_16_1_1F)
+        assert 'CompressedSamples^MR1' == ds.PatientName
+        assert 8192 == len(ds.PixelData)
+
+        # Implicit little
+        ds = dcmread(IMPL_16_1_1F)
+        assert 'CompressedSamples^MR1' == ds.PatientName
+        assert 8192 == len(ds.PixelData)
+
+        # Deflated little
+        ds = dcmread(DEFL_8_1_1F)
+        assert '^^^^' == ds.PatientName
+        assert 262144 == len(ds.PixelData)
+
+        # Explicit big
+        ds = dcmread(EXPB_16_1_1F)
+        assert 'CompressedSamples^MR1' == ds.PatientName
+        assert 8192 == len(ds.PixelData)
+
+    @pytest.mark.parametrize("fpath,data", REFERENCE_DATA_UNSUPPORTED)
+    def test_can_access_unsupported_dataset(self, fpath, data):
+        """Test can read and access elements in unsupported datasets."""
```

Comment:
Are these testing any new code?  Seems like they are likely a repeat of existing tests as they don't access related data elements?

Reply:
Yeah, I tend to over test, I'll trim it down.